Eevee.Sleep.Bot/Handlers/InteractionHandler.cs

```diff
@@ -126,14 +126,19 @@ await interaction
 
         await component.RespondAsync("You have been given the role!", ephemeral: true);
 
-        var roleId = ulong.Parse(customId.Replace("role", ""));
-        var user = (SocketGuildUser)component.User;
+        if (component.User is not null and SocketGuildUser) {
+            var user = (SocketGuildUser)component.User;
+            await user.RemoveRolesAsync(
+                user.Roles
+                    .Where(r => !ExcludedRoleNames.RoleNames.Contains(r.Name))
+                    .Select(r => r.Id).ToArray()
+            );
+
+            var roleId = ulong.Parse(customId.Replace("role", ""));
+            await user.AddRoleAsync(roleId);
+        } else {
+            throw new ArgumentException("User is not a guild user!");
```

Comment:
Can just skip processing.

Reply:
The messages are sending earlier is "Your request has been accepted!" (`await component.RespondAsync("You have been given the role!", ephemeral: true)`) and then change the message behind it to "Couldn't do it due to an error"

Reply:
Deleted it while I made serializer
[042bf5a](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/042bf5ae19def3a76b4fa219af2021bdffdd459e)